pkg/gcc/adaptive_threshold.go

```diff
@@ -0,0 +1,60 @@
+package gcc
+
+import (
+	"time"
+)
+
+type adaptiveThresholdOption func(*adaptiveThreshold)
+
+func setInitialThreshold(t time.Duration) adaptiveThresholdOption {
+	return func(at *adaptiveThreshold) {
+		at.thresh = t
+	}
+}
+
+type adaptiveThreshold struct {
+	thresh              time.Duration
+	overuseCoefficientU float64
+	overuseCoefficientD float64
+	min                 time.Duration
+	max                 time.Duration
+}
+
+func newAdaptiveThreshold(opts ...adaptiveThresholdOption) *adaptiveThreshold {
```

Comment:
Could you write a short `godoc` on what a AdaptiveThreshold does? Nothing crazy, just to re-inforce what it does to someone reading the code?